File: 890-lemonade-change/lemonade-change.cpp
class Solution {
public:
    bool lemonadeChange(vector<int>& bills) 
    {
        int fc = 0, tc = 0, twc = 0;
        for(auto it: bills)
        {
            if(it == 5) fc++;
            else if(it == 10){
                tc++;
                if(fc > 0) fc--;
                else return false;
            }
            else if(it == 20){
                twc++;
                if(tc > 0) {
                    tc--;
                    if(fc > 0){
                        fc--;
                    }
                    else return false;
                }
                else if(fc >= 3) {
                    fc -= 3;
                }
                else return false;
            }
        }    
        return true;
    }
};

File: 890-lemonade-change/lemonade-change.py
class Solution:
    def lemonadeChange(self, bills: list[int]) -> bool:
        fc = 0
        tc = 0 
        twc = 0

        for bill in bills:
            if bill == 5:
                fc += 1
            elif bill == 10:
                tc += 1
                if fc > 0:
                    fc -= 1
                else:
                    return False
            elif bill == 20:
                twc += 1
                if tc > 0:
                    tc -= 1
                    if fc > 0:
                        fc -= 1
                    else:
                        return False
                elif fc >= 3:
                    fc -= 3
                else:
                    return False
            else:
                return False
            
        return True